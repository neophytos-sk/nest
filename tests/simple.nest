with_mode decl eval {
    multiple varchar words
    varint count
}

multiple words {abc def ghi}
count 123
