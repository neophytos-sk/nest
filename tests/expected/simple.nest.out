<lang x-nsp="::nest::lang">
    <decl x-name="base_type" x-type="meta" x-id="base_type"/>
    <decl x-name="varchar" x-type="base_type" x-id="varchar"/>
    <decl x-name="bool" x-type="base_type" x-id="bool"/>
    <decl x-name="varint" x-type="base_type" x-id="varint"/>
    <decl x-name="byte" x-type="base_type" x-id="byte"/>
    <decl x-name="int16" x-type="base_type" x-id="int16"/>
    <decl x-name="int32" x-type="base_type" x-id="int32"/>
    <decl x-name="int64" x-type="base_type" x-id="int64"/>
    <decl x-name="double" x-type="base_type" x-id="double"/>
    <decl x-name="struct" x-type="meta" x-id="struct">
        <decl x-name="id" x-type="varchar"/>
        <decl x-name="name" x-type="varchar"/>
        <decl x-name="type" x-type="varchar"/>
        <decl x-name="nsp" x-type="varchar"/>
        <decl x-name="default_value" x-type="varchar" x-default_value=""/>
        <decl x-name="slot" x-type="struct" x-id="struct.slot" x-default_value="" x-container="multiple">
            <decl x-name="id" x-type="varchar"/>
            <decl x-name="name" x-type="varchar"/>
            <decl x-name="type" x-type="varchar"/>
            <decl x-name="default_value" x-type="varchar" x-default_value=""/>
            <decl x-name="optional_p" x-type="bool" x-default_value="false"/>
            <decl x-name="container" x-type="varchar" x-default_value=""/>
        </decl>
        <decl x-name="pk" x-type="varchar"/>
        <decl x-name="is_final_if_no_scope" x-type="bool"/>
    </decl>
    <decl x-name="fun" x-type="struct" x-id="fun">
        <decl x-name="name" x-type="varchar"/>
        <decl x-name="param" x-type="varchar" x-default_value="" x-container="multiple"/>
        <decl x-name="body" x-type="varchar"/>
    </decl>
</lang>

<nest>
    <decl x-name="words" x-type="varchar" x-container="multiple"/>
    <decl x-name="count" x-type="varint"/>
    <inst x-name="words" x-type="varchar" x-container="multiple">abc</inst>
    <inst x-name="words" x-type="varchar" x-container="multiple">def</inst>
    <inst x-name="words" x-type="varchar" x-container="multiple">ghi</inst>
    <inst x-name="count" x-type="varint">123</inst>
</nest>

